File: l33tcode/two-sum.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> tuple[int, int]:
        num_to_pos = {}
        for i, num in enumerate(nums):
            second = target - num
            if second in num_to_pos:
                return i, num_to_pos[second]

            num_to_pos[num] = i

        raise ValueError("Not found")

File: l33tcode_rs/src/two_sum.rs
#[allow(dead_code)]
struct Solution;

use std::collections::HashMap;

impl Solution {
    #[allow(dead_code, clippy::needless_pass_by_value)]
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map: HashMap<i32, i32> = HashMap::new();

        for (i, num) in nums.iter().enumerate() {
            if let Some(pos) = map.get(&(target - num)) {
                return vec![*pos, i.try_into().unwrap()];
            }

            map.insert(*num, i.try_into().unwrap());
        }

        vec![]
    }
}
